xarray/tests/test_namedarray.py

```diff
@@ -0,0 +1,116 @@
+import numpy as np
+import pytest
+
+from xarray.namedarray.core import NamedArray, as_compatible_data
+from xarray.namedarray.utils import T_DuckArray
+
+
+@pytest.fixture
+def random_inputs():
+    return np.random.random((3, 4, 5)).astype(np.float32)
+
+
+@pytest.mark.parametrize(
+    "input_data, expected_output",
+    [
+        ([1, 2, 3], np.array([1, 2, 3])),
+        (np.array([4, 5, 6]), np.array([4, 5, 6])),
+    ],
+)
+def test_as_compatible_data(input_data: T_DuckArray, expected_output: T_DuckArray):
```

Comment:
need to test errors for masked array, named array, conversion of tuple to object array.